icechunk/src/session.rs

```diff
@@ -1162,20 +1272,22 @@ async fn verified_node_chunk_iterator<'a>(
     snapshot_id: &'a SnapshotId,
     change_set: &'a ChangeSet,
     node: NodeSnapshot,
+    extent: Option<ManifestExtents>,
```

Comment:
Important: All iterators have gain `Option<ManifestExtents>` to do some filtering to extents=. This is where I'll need to add lifetimes to switch to `Option<&ManifestExtents>`

Reply:
I wonder if for all these cases, instead of using `Option` it's easier to introduce a (constant) match all ManifestExtents: `ManifestExtents::all : ManifestExtents`